src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -425,17 +444,26 @@ private void WorkerBodyLoop()
                 catch (Exception exc)
                 {
                     eventLoopError = exc;
-                    throw;
+                    if (!(exc is HandlerShutdownException))
```

Comment:
That's an awful lot of code in the `try` block; this "swallow any HandlerShutdownException" logic makes me nervous that there's some point in the `try` where this can get thrown, and some state will be left half-updated.  I'd rather see local `try`/`catch` blocks just around the calls that are expected to throw, so it's easier to verify that all relevant state is left consistent.